modules/bezug_rct2/rct_read_bezug.py

```diff
@@ -27,10 +26,8 @@ def writeRam(fn, val, rctname):
 #
 # Entry point with parameter check
 #
-
-
-def main(argv: List[str]):
-    rct_lib.init(argv[0])
+def main():
+    rct_lib.init(sys.argv)
 
     clientsocket = rct_lib.connect_to_server()
```

Comment:
Ich sehe hier ein großes Problem. Es wird nur einmal geprüft, ob der clientsocket erstellt wurde. Wenn dann in dem if-Block irgendwas passiert, wird der Socket nicht mehr geschlossen. Das ist auch in den anderen rct_read_* so umgesetzt.

Für solche Fälle gibt es in Python ContextHandler. Das ist jedoch eine größere Änderung, da die rct_lib in eine Klasse umgeschrieben werden muss, welche dann __enter__() und __exit__() implementiert. Die Anpassungen möchte ich nicht machen, ohne ein Testsystem zu haben. Kannst Du Dir das mal ansehen?

Reply:
Das ist ja bei der 1.-9'er kein Problem.
Wenn eine Abfrage abkackt, was solls. es wird ja nach 10 Sekunden einen neue Verbindug aufgebaut.
Da kann sich  nichts ansammeln.

Bei der 2.0 ist das natürlich anders.
Wenn ich das richtig Verstehe sind dann alle Python-Module-Scripte dauerhaft geladen und dürfen natürlich
keinen "Müll" erzeugen der nicht weggeräumt wird.

Reply:
Hmmm der LRS hält ja auch die python Scripte im Speicher.
Also gleiches Problem wie bei der 2.0
ALle Scripte müssen "langlauffest" sein.

Reply:
> `Für solche Fälle gibt es in Python ContextHandler. Das ist jedoch eine größere Änderung, da die rct_lib in eine Klasse umgeschrieben werden muss, welche dann enter() und exit() implementiert. Die Anpassungen möchte ich nicht machen, ohne ein Testsystem zu haben. Kannst Du Dir das mal ansehen?

Reicht es nicht einen Try-Except  oder Try-Finally oder wie das bei Python heissen mag, einzusetzen
und das Socket auch im Fehlerfall zu schliessen.